bigchaindb/core.py

```diff
@@ -411,11 +411,30 @@ def get_owned_ids(self, owner):
                     # subfulfillment for `owner`
                     if utils.condition_details_has_owner(output['condition']['details'], owner):
                         tx_link = TransactionLink(tx['id'], index)
-                # check if input was already spent
-                if not self.get_spent(tx_link.txid, tx_link.output):
-                    owned.append(tx_link)
+                links.append(tx_link)
+        return links
+
+    def get_owned_ids(self, owner):
+        """Retrieve a list of ``txid`` s that can be used as inputs.
+
+        Args:
+            owner (str): base58 encoded public key.
+
+        Returns:
+            :obj:`list` of TransactionLink: list of ``txid`` s and ``output`` s
+            pointing to another transaction's condition
+        """
+        return self.get_outputs_filtered(owner, include_spent=False)
 
-        return owned
+    def get_outputs_filtered(self, owner, include_spent=True):
```

Comment:
There's a school of thought that suggests that functions should do one thing. This was the idea behind separating the filtering from the fetching.